ocaml/parsing/jane_syntax.mli

```diff
@@ -72,7 +62,8 @@ module Comprehensions : sig
           (only allowed with [-extension immutable_arrays]) *)
 
   val expr_of :
-    loc:Location.t -> expression -> Parsetree.expression_desc With_attributes.t
+    loc:Location.t -> ?attrs:Parsetree.attributes ->
```

Comment:
I'm sympathetic here. The reason I opted for the optional parameter is that, really, ast_mapper should be the only place that uses the parameter. The `expr_of` function really has no business dealing with attributes at all -- it's really that the structure in the parsetree is a little wrong. Really, there should be a `Pexp_attributes` constructor of `expression_desc` that wraps an expression with attributes. Or maybe even just one attribute at a time (I'd actually prefer that). This would just be so much better, but it's not what we have.

All that said, let me see what I can do, especially after rebasing over your other patch.